includes/class-wp-statistics-install.php

```diff
@@ -411,19 +411,26 @@ public function plugin_upgrades()
          *
          * @version 13.0.0
          */
-        if (!DB::isColumnType('visitor', 'ID', 'bigint(20)')) {
+        /*
+         * MySQL since version 8.0.19 doesn't honot  display width specification
+         * so we have to handle accept BIGINT(20) and BIGINT.
+         *
+         * see: https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-19.html  
+         * - section Deprecation and Removal Notes
+         */
+        if (!DB::isColumnType('visitor', 'ID', 'bigint(20)') && !DB::isColumnType('visitor', 'ID', 'bigint')) {
```

Comment:
It seems like the operand should be `OR`, right? This implies that if the column type is not `bigint(20)` or `bigint`, then we alter the table, doesn't it?

Reply:
I don't think so. The operand should be AND in my opinion...

### Lets consider situation when the column has type of  `bigint` .

* `!DB::isColumnType('visitor', 'ID', 'bigint(20)')` is TRUE
* and `!DB::isColumnType('visitor', 'ID', 'bigint')` is FALSE

When we use AND, the whole condition is going to be `FALSE` - (`FALSE AND TRUE`) and it would not trigger the migration. This is OK.

When we use OR, the whole condition is going to be `TRUE` ( `FALSE OR TRUE` ) and it would trigger the migration, which is incorrect. The column type is already in enabled types.

### Lets consider situuation with column type `int`

* `!DB::isColumnType('visitor', 'ID', 'bigint(20)')` is now TRUE
* and `!DB::isColumnType('visitor', 'ID', 'bigint')` is TRUE

In this case it is not important whether we are using AND or OR, as it would trigger migration in both cases.

## Is there a better way?

Yes, more clean way of handling this would be probably changing signature `isColumnType`  to enable list of permitted types. Unfortunately I don't have enough PHP skills for that. 

I also don't know whether the `isColumnType` is used somewhere else.

Reply:
Ah, I got it; thanks for clarifying.

Let's maintain the current functionality of `isColumnType` without any modifications.